src/epics/ipfs/uploadToIPFS.epic.js

```diff
@@ -0,0 +1,64 @@
+import { from, of } from 'rxjs';
+import { switchMap, withLatestFrom, map, catchError, tap } from 'rxjs/operators';
+import { ofType } from 'redux-observable';
+import { getSelectedProjectId } from '../../selectors/projects';
+import { ipfsActions } from '../../actions';
+
+/**
+ * Add a timestamp to the upload object we are about to save in order to have the possibility to build a timeline
+ * of backups
+ * @param {string} shareURL The share URL given to the user
+ */
+const addTimeStamp = (shareURL) => {
+    return { shareURL: shareURL, timestamp: Date.now() }
+}
+
+/**
+ * This is needed becase atm the file explorer has no way to auto update itself when the
+ * underlying state has actually changed (per example by using the backend.js class)
+ */
+const updateFileSystemState = (project) => {
```

Comment:
I think this function should return a promise which is resolved when the callback from `getChildren` is called.

Reply:
This should be fixed in the latest commit